server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTasksLogger.java

```diff
@@ -97,4 +106,12 @@ private void logTopResourceConsumingQueries() {
             SEARCH_TASK_DETAILS_LOGGER.info(new SearchShardTaskDetailsLogMessage(topQuery.v2()));
         }
     }
+
+    void setLogTopQueriesSize(int topQueriesSize) {
+        this.topQueriesSize = topQueriesSize;
```

Comment:
This is used in two places in the `recordSearchTask` method. Now that we are making it dynamic, it is prone to race-conditions. Can we please fix?

Reply:
Sure, will make it Atomic in the next commit.